solver/src/driver.rs

```diff
@@ -373,45 +350,59 @@ impl Driver {
             .context("failed to estimate gas price")?;
         tracing::debug!("solving with gas price of {}", gas_price_wei);
 
-        let settlements = self
-            .run_solvers(liquidity, gas_price_wei)
+        let mut solver_settlements = Vec::new();
+
+        for (solver_id, settlements) in self
+            .solvers
+            .run(
+                liquidity,
+                gas_price_wei,
+                self.solver_time_limit,
+                Some(self.metrics.as_ref()),
+            )
             .await
-            .filter_map(solver_settlements::filter_bad_settlements)
-            .inspect(|(name, settlements)| {
-                for settlement in settlements.iter() {
-                    tracing::debug!("solver {} found solution:\n {:?}", name, settlement);
```

Comment:
Are we removing this log then? I think we might still want to be informed when solutions are found.